lamindb/_record.py

```diff
@@ -847,6 +864,32 @@ def check_name_change(record: Record):
                 raise RecordNameChangeIntegrityError
 
 
+def check_key_change(record: Record):
+    """Errors if a record's key has falsely changed."""
+    if not record.pk or record.key is None or not hasattr(record, "_key"):
+        return
+
+    old_key = record._key
+    new_key = record.key
+
+    old_key_suffix = (
+        record.suffix
```

Comment:
Transform does not have `suffix`.

Reply:
I'd only add a consistency check for artifact at the very end to ensure that the new suffix is still consistent.

For transforms this isn't necessary because we don't have the suffix column.